51_fl_diff_shipment.py

```diff
@@ -0,0 +1,258 @@
+# importing the required libraries
+import pandas as pd
+import numpy as np
+import altair as alt
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# loading data for drug prescriptions in Florida and neighbouring states
+prescriptions = pd.read_csv("05_cleaned_data/prescriptions_fl.csv", low_memory=False)
+prescriptions
+
+# reducing the number of columns in the drug prescriptions dataset by including only the relevant attributes to create a new dataset
+prescriptions_reduced = prescriptions[
+    [
+        "DRUG_CODE",
+        "DRUG_NAME",
+        "QUANTITY",
+        "UNIT",
+        "STRENGTH",
+        "CALC_BASE_WT_IN_GM",
+        "DOSAGE_UNIT",
+        "Product_Name",
+        "Ingredient_Name",
+        "Measure",
+        "MME_Conversion_Factor",
+        "dos_str",
+        "Year",
+        "Month",
+        "StateFIPS",
+        "StateName",
+        "CountyFIPS",
+        "state_abbrev",
+        "FIP_unique",
+        "Population",
+        "county_test",
+    ]
+]
+# creating a copy of reduced dataset of prescriptions and converting some of the attributes to appropriate data type
+
+prescriptions_reduced_copy = prescriptions_reduced.copy()
+
+prescriptions_reduced_copy["Year"] = prescriptions_reduced_copy["Year"].astype("int64")
+prescriptions_reduced_copy["DRUG_CODE"] = prescriptions_reduced_copy[
+    "DRUG_CODE"
+].astype("int64")
+prescriptions_reduced_copy["Month"] = prescriptions_reduced_copy["Month"].astype(
+    "int64"
+)
+prescriptions_reduced_copy["Population"] = prescriptions_reduced_copy[
+    "Population"
+].astype("int64")
+
+# creating a dataset that has all the drug prescriptions in the state of Florida
+
+florida_prescriptions = prescriptions_reduced_copy[
+    prescriptions_reduced_copy["StateName"] == "Florida"
+]
+# florida_prescriptions["shipment_per_100k"] = (florida_prescriptions["QUANTITY"] / florida_prescriptions["Population"]) * 100_000
+
+florida_prescriptions["shipment_per_100k"] = (
+    (
+        florida_prescriptions["dos_str"]
+        * florida_prescriptions["DOSAGE_UNIT"]
+        * florida_prescriptions["MME_Conversion_Factor"]
+    )
+    / (florida_prescriptions["Population"])
+    * 100000
+)
+florida_prescriptions
+# calculating and displaying total number of drug prescriptions that took place in Florida. results are grouped and displayed year and county wise
+
+florida_prescriptions_result = (
+    florida_prescriptions.groupby(["Year", "county_test"])["shipment_per_100k"]
+    .sum()
+    .reset_index()
+)
```

Comment:
use florida_prescriptions_copy in the groupby formula, after you have made copy of florida_prescriptions,